presto-kafka/src/main/java/io/prestosql/plugin/kafka/confluent/ConfluentSchemaRegistryTableDescriptionSupplier.java

```diff
@@ -0,0 +1,383 @@
+        return Optional.of(new KafkaTopicDescription(tableName, Optional.of(schemaTableName.getSchemaName()), topic, key, message));
+    }
+
+    private KafkaTopicFieldGroup getFieldGroup(AvroSchemaConverter avroSchemaConverter, String subject)
+    {
+        try {
+            Schema schema = new Schema.Parser().parse(schemaRegistryClient.getLatestSchemaMetadata(subject).getSchema());
+            List<Type> types = avroSchemaConverter.convertAvroSchema(schema);
+            ImmutableList.Builder<KafkaTopicFieldDescription> fieldsBuilder = ImmutableList.builder();
+            if (schema.getType() != Schema.Type.RECORD) {
+                checkState(types.size() == 1, "incompatible schema");
+                fieldsBuilder.add(new KafkaTopicFieldDescription(
+                        subject,
+                        getOnlyElement(types),
+                        subject,
+                        null,
+                        null,
+                        null,
+                        false));
+            }
+            else {
+                List<Schema.Field> avroFields = schema.getFields();
+                checkState(avroFields.size() == types.size(), "incompatible schema");
+
+                for (int i = 0; i < types.size(); i++) {
+                    Schema.Field field = avroFields.get(i);
+                    fieldsBuilder.add(new KafkaTopicFieldDescription(
+                            field.name(),
+                            types.get(i),
+                            field.name(),
+                            null,
+                            null,
+                            null,
+                            false));
+                }
+            }
+            return new KafkaTopicFieldGroup(AvroRowDecoderFactory.NAME, Optional.empty(), getDecoderParams(subject), fieldsBuilder.build());
+        }
+        catch (IOException | RestClientException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Unable to get field group for '%s' subject"), e);
+        }
+    }
+
+    private static Optional<Map<String, String>> getDecoderParams(String subject)
+    {
+        return Optional.of(ImmutableMap.<String, String>builder()
+                .put(SUBJECT, subject)
+                .putAll(DEFAULT_DECODER_PARAMS)
+                .build());
+    }
+
+    // If the default subject naming strategy is not used, the key and message subjects
+    // can be specified by adding them to the table name as follows:
+    //  <tablename>&key-subject=<key subject>&message-subject=<value subject>
+    // ex. kafka.default."mytable&key-subject=foo&message-subject=bar"
+    @VisibleForTesting
+    static TopicAndSubjects parseTopicAndSubjects(SchemaTableName encodedSchemaTableName)
+    {
+        String encodedTableName = encodedSchemaTableName.getTableName();
+        List<String> parts = Splitter.on(DELIMITER).trimResults().splitToList(encodedTableName);
```

Comment:
Instead of splitting the String, can we define a Pattern which captures required topic, key subject and message subject